sonar-db/src/main/java/org/sonar/db/qualityprofile/ActiveRuleDao.java

```diff
@@ -206,4 +210,17 @@ private ParamIdToDto(ActiveRuleMapper mapper) {
       return mapper.selectParamsByActiveRuleIds(partitionOfIds);
     }
   }
+
+  private static class RuleIdToDto implements Function<List<Integer>, List<ActiveRuleDto>> {
+    private final ActiveRuleMapper mapper;
+
+    private RuleIdToDto(ActiveRuleMapper mapper) {
+      this.mapper = mapper;
+    }
+
+    @Override
+    public List<ActiveRuleDto> apply(@Nonnull List<Integer> partitionOfRuleIds) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The "partitionOfRuleIds" parameter nullability is different in the superclass method, and that should not be changed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)